### PR TITLE
[SCRUM-26] event processor created for order and orderShoe in product…

### PR DESCRIPTION
**Pull request by evannwang**
…ion, models.py modified

### CODE REVIEW

File: backend-python/event_processor.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
+
+from helperclass import *
+from models import *
+from app_config import db
+
+
+        
+OrderStatusNameList = ["订单创建", "打样单创建", "打样单初步确认","打样单分配","打样单修改","打样单最终确认",
```
[the remark this answers is not in the document]
Comment (evannwang) [reply backend-python/event_processor.py:8]: done

File: backend-python/event_processor.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        self.orderStatusNameList = OrderStatusNameList
+        self.orderShoeStatusNameList = OrderShoeStatusNameList
+        ### idToNode dict
+        self.orderStatusidToNode = dict()
+        self.orderShoeStatusidToNode = dict()
+        ### Procedures
+        self.orderShoeLogisticProcedures = None
+        self.orderShoeProductionProcedures = None
+        self.orderProcedures = None
+        self.constructProcedures()
```
[the remark this answers is not in the document]
Comment (evannwang) [reply backend-python/event_processor.py:76]: done

File: backend-python/event_processor.py [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+        for i in range(len(self.orderStatusNameList)):
+            cur_status = OrderStatusVal(i,self.orderStatusNameList[i])
+            node_insert = LinkedListNode(val = cur_status)
+            self.orderStatusidToNode[i] = node_insert
+            if i == 0:
+                self.orderStatusPath.setHead(node_insert)
+                prev = node_insert
+            else:
+                prev.setNext(node_insert)
+                node_insert.setPrev = prev
```
[the remark this answers is not in the document]
Comment (evannwang) [reply backend-python/event_processor.py:92]: done

File: backend-python/event_processor.py [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+            statusName = OrderShoeStatusNameList[i]
+            node = GraphNode(val = OrderShoeStatusVal(i,statusName))
+            self.orderShoeStatusidToNode[i] = node
+            prevs = []
+            if i == 0:
+                self.orderShoeProductionPath.setSource(node)
+            elif i == 9:
+                prevs.append(self.orderShoeStatusidToNode[3])
+            elif i == 16:
+                prevs = [prev] + [self.orderShoeStatusidToNode[8]]
```
[the remark this answers is not in the document]
Comment (evannwang) [reply backend-python/event_processor.py:110]: done

File: backend-python/event_processor.py [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+            entity = (db.session.query(OrderShoeStatus).
+                      filter(OrderShoeStatus.order_shoe_id == event.event_order_shoe_id)
+                      .first())
+            result = entity != None
+        return result
+    
+    def processOrderEvent(self, event, operation):
+        order_id = event.event_order_id
+        modified_value = operation.operation_modified_value
+        modifed_status = operation.operation_modified_status
```
[the remark this answers is not in the document]
Comment (evannwang) [reply backend-python/event_processor.py:207]: done

File: backend-python/event_processor.py [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+        return result
+
+    def processOrderShoeEvent(self, event, operation):
+        order_shoe_id = event.event_order_shoe_id
+        cur_stat ,cur_val= self.dbQueryOrderShoeStatus(order_shoe_id)
+        modified_status = operation.operation_modified_status
+        modified_value = operation.operation_modified_value
+        print("current order_shoe_id is " + str(order_shoe_id))
+        print("current status is " + str(cur_stat) + " with value " + str(cur_val))
+        print("modifiying into " + str(modified_status) + " with value " + str(modified_value))
```
[the remark this answers is not in the document]
Comment (evannwang) [reply backend-python/event_processor.py:240]: done

File: backend-python/event_processor.py [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+            ### TODO INSERT EVENT INTO DB
+        else:
+            print("Modifying an non exist status or existing status with a wrong value")
+            result = False
+        return result
+
+
+
+
+    def processEvent(self, event):
```
[the remark this answers is not in the document]
Comment (evannwang) [reply backend-python/event_processor.py:259]: done

File: backend-python/event_processor.py [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+                    db.session.add(new_entity)
+                    db.session.commit()
+                else:
+                    entity = (db.session.query(OrderShoeStatus).
+                                filter(OrderShoeStatus.order_shoe_id == event.event_order_shoe_id ,
+                                       OrderShoeStatus.current_status == operation.operation_modified_status).
+                                       first())
+                    entity.current_status = next_status[0]
+                    entity.current_status_value = 0
+                    db.session.commit()
```
[the remark this answers is not in the document]
Comment (evannwang) [reply backend-python/event_processor.py:311]: done

**Review by GuikangZhong — APPROVED** [review]

**Pull request merged**